sdk/search/azure-search-documents/azure/search/documents/_internal/_search_client.py

```diff
@@ -75,6 +78,7 @@ class SearchClient(HeadersMixin):
     def __init__(self, endpoint, index_name, credential, **kwargs):
         # type: (str, str, AzureKeyCredential, **Any) -> None
 
+        get_api_version(kwargs, "2019-05-06-Preview")
```

Comment:
So I guess here it's used has a "check", then I would split a "check" and a "get" into two different methods, with "get" calling check first. I find it easier to read intent

Reply:
It's optional feedback, I let you decide if that makes the code less efficient